### ISSUE TITLE
[Debugger] NullReferenceException when print non-printable elements in STL container.

### ISSUE THREAD

**Issue by lixin-wei** [microsoft/vscode-cpptools#7324]
Type: Debugger

# System Info
I'm using the remote mode of vscode.
- OS and Version:
    - Client: MacOS Big Sur 11.2.3 (20D91)
    - Remote Host: centOS 7, Linux 4.9.151
- VS Code Version: 
```
Version: 1.55.0
Commit: c185983a683d14c396952dd432459097bc7f757f
Date: 2021-03-30T16:01:05.981Z (1 wk ago)
Electron: 11.3.0
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Darwin x64 20.3.0
```
- C/C++ Extension Version: v1.3.0-insiders5
- gdb version: GNU gdb (GDB) Red Hat Enterprise Linux 7.12.1

# Bug Description
Just look at the following screenshot.

When I try to view an STL container with non-printable elements, a NullReferenceException occurred and the debugger is crashed.

I'm sure it's vscode's bug because printing this container by `-exec p` is OK.

![image](https://user-images.githubusercontent.com/19600697/114125356-dff44000-9928-11eb-8770-4581f916c326.png)

# To Reproduce

1. compile and debug the sample code using my task.json and launch.json.
2. try to view STL container `mp`.

sample code:
``` cpp
#include <unordered_map>
#include <map>
#include <cstdio>
using namespace std;

class Key {
public:
    int x;
    Key(int x) {
        this->x = x;
    }
    friend bool operator < (const Key& a, const Key& b) {
        return a.x < b.x;
    }
};
int main() {
    map<Key, int> mp;
    mp[Key(0)]=0; mp[Key(1)]=1;
    printf("%d\n", mp[Key(0)]);
}
```

launch.json

```
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "name": "g++ - Build and debug active file",
            "type": "cppdbg",
            "request": "launch",
            "program": "${fileDirname}/${fileBasenameNoExtension}",
            "args": [],
            "stopAtEntry": false,
            "cwd": "${workspaceFolder}",
            "environment": [],
            "externalConsole": false,
            "MIMode": "gdb",
            "setupCommands": [
                {
                    "description": "Enable pretty-printing for gdb",
                    "text": "-enable-pretty-printing",
                    "ignoreFailures": true
                }
            ],
            "preLaunchTask": "C/C++: g++ build active file",
            "miDebuggerPath": "/usr/bin/gdb"
        }
    ]
}
```

tasks:
```
{
    "tasks": [
        {
            "type": "cppbuild",
            "label": "C/C++: g++ build active file",
            "command": "/usr/bin/g++",
            "args": [
                "-g",
                "${file}",
                "-o",
                "${fileDirname}/${fileBasenameNoExtension}"
            ],
            "options": {
                "cwd": "${workspaceFolder}"
            },
            "problemMatcher": [
                "$gcc"
            ],
            "group": {
                "kind": "build",
                "isDefault": true
            },
            "detail": "Task generated by Debugger."
        }
    ],
    "version": "2.0.0"
}
```

**Comment by lixin-wei** [comment]
could anyone have a look?

**Comment by WardenGnaw** [comment]
This is due to https://github.com/microsoft/MIEngine/commit/b774ad486dc0e8a07d956df3a6bb51813971104a#diff-ac0f039773063f8f8697d4ab9f9d6158d2e9069366abdf9f8b86e991b8fd255b

**Comment by lixin-wei** [comment]
> This is due to not handling nulls in memoryContext's.
> 
> Fixed in [microsoft/MIEngine@b774ad4#diff-ac0f039773063f8f8697d4ab9f9d6158d2e9069366abdf9f8b86e991b8fd255b](https://github.com/microsoft/MIEngine/commit/b774ad486dc0e8a07d956df3a6bb51813971104a#diff-ac0f039773063f8f8697d4ab9f9d6158d2e9069366abdf9f8b86e991b8fd255b)

great, when will you apply this fix and release a new version so I can update it?

**Comment by lixin-wei** [comment]
@WardenGnaw BTW, I'm wondering how MIEngine is used in vscode? As far as I known, MIEngine is written in C#, how C# projects can be run in linux?  And I also know that MIEngine is a plugin of VS, how it can be used in vscode?

**Comment by Delta-in-hub** [comment]
Same issue.

```
版本: 1.55.2 (user setup)
提交: 3c4e3df9e89829dce27b7b5c24508306b151f30d
日期: 2021-04-13T09:35:57.887Z
Electron: 11.3.0
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Windows_NT x64 10.0.19042

C\C++ 1.3.0

GNU gdb (GDB) 8.3.1
```

**Comment by Delta-in-hub** [comment]
> @WardenGnaw BTW, I'm wondering how MIEngine is used in vscode? As far as I known, MIEngine is written in C#, how C# projects can be run in linux? And I also know that MIEngine is a plugin of VS, how it can be used in vscode?

Have you solved this issue?

**Comment by lixin-wei** [comment]
@Delta-in-hub They've fixed it. Just wait for their release.

**Comment by lixin-wei** [comment]
@WardenGnaw Hi, when will you apply this fix and release a new version so I can update it?

**Comment by WardenGnaw** [comment]
We unfortunately did not make it into the most recent release (1.3.1). We are hoping to ship an update to the debugger in the next preview.

Sorry for the delay.

**Comment by lixin-wei** [comment]
> We unfortunately did not make it into the most recent release (1.3.1). We are hoping to ship an update to the debugger in the next preview.
> 
> Sorry for the delay.

Thanks, please ping me when you release this fix.

**Comment by Colengms** [comment]
This fix should be in 1.4.0.
